app.rb

```diff
@@ -0,0 +1,123 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+    @classroom = Classroom.new('Grade 10')
+  end
+
+  def list_all_books
+    if @books.empty?
+      puts 'Oops! No book, please add a book'
+      return
+    end
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    if @people.empty?
+      puts 'Oops! No people, please add people'
+      return
+    end
+    @people.each do |person|
+      puts "[#{person.class}] Name: #{person.name} ID: #{person.id} Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a student or a teacher?'
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid! Please enter 1 or 2'
+    end
+  end
+
+  def create_student
+    puts 'Name:'
+    name = gets.chomp
+
+    print 'Age:'
+    age = gets.chomp.to_i
+
+    print 'Has parent permission? [Y/N]'
+    parent_permission = gets.chomp.downcase
+
+    student = Student.new(age, @classroom, name, parent_permission: parent_permission == 'y')
+    @people << student
+    puts 'Student created successfully'
+  end
+
+  def create_teacher
+    puts 'Name:'
+    name = gets.chomp
+
+    print 'Age:'
+    age = gets.chomp.to_i
+
+    print 'Specialization:'
+    specialization = gets.chomp
+
+    @people << Teacher.new(age, specialization, name)
+    puts 'Teacher created successfully'
+  end
```

Comment:
- Hey, when trying to create a student, I get this error of a number of arguments this is coming from your constructor in the teacher.rb file. please, kindly fix this.
- Also, when creating a person, kindly specify the number to indicate a person. e.g. `Do you want to create a teacher or student? enter 2 for the teacher, 1 for the student.` 


![Screenshot from 2023-04-28 21-21-29](https://user-images.githubusercontent.com/59397159/235248026-f7da2080-1058-4eb4-a2bb-adf2c1cb96b7.png)

Reply:
NOTE: Since we can't create a person, we can't access other methods in your app except the 
`create a book and List all books`. Let's fix this error and ensure the app works fine before submitting for another review.